addie/advanced/isrp_gui_table_init.py

```diff
@@ -0,0 +1,232 @@
+            self.err_messenger("'Sample-2 Scans' missing!")
+            return 
+        if not self.parent.ui.secondary_scattering_ratio.text().strip():
+            self.err_messenger("'Secondary Scattering Ratio' missing!")
+            return 
+        cond1 = self.parent.ui.plazcek_fit_range_min.text().strip()
+        cond2 = self.parent.ui.plazcek_fit_range_max.text().strip()
+        if not cond1 or not cond2:
+            self.err_messenger("Plazcek info incomplete!")
+            return 
+        cond1 = self.parent.ui.subs_init.text().strip()
+        cond2 = self.parent.ui.subs_rep.text().strip()
+        if not cond1 or not cond2:
+            self.err_messenger("Substitution info incomplete!")
+            return 
+        cond1 = self.parent.ui.ft_qrange.text().strip()
+        cond2 = self.parent.ui.ff_rrange.text().strip()
+        cond3 = self.parent.ui.ff_qrange.text().strip()
+        if not cond1 or not cond2 or not cond3:
+            self.err_messenger("Fourier transform info incomplete!")
+            return 
+        _current_folder = self.parent.parent.current_folder
+        working_dir = QFileDialog.getExistingDirectory(self.parent,
+                                                       "Select Working Directory",
+                                                       _current_folder,
+                                                       QFileDialog.ShowDirsOnly)
+
+        if not working_dir:
+            return
+
+        try:
+            fod_output = open(os.path.join(working_dir, "fod.inp"), "w")
+        except IOError:
+            self.err_messenger("Permission denied! Choose another working folder!")
+            return
+
+        fod_output.write(self.parent.ui.sample_1_title.text() + "  # sample_1_title\n")
+        fod_output.write(self.parent.ui.sample_2_title.text() + "  # sample_2_title\n")
+        fod_output.write(self.parent.ui.bkg_title.text() + "  # background_title\n")
+        fod_output.write(self.parent.ui.bkg_scans.text() + "  # background scannrs\n")
+        fod_output.write(self.parent.ui.sample_1_scans.text() + "  # sample_1_scannrs\n")
+        fod_output.write(self.parent.ui.sample_2_scans.text() + "  # sample_2_scannrs\n")
+        fod_output.write(self.parent.ui.subs_init.text() + " / " + self.parent.ui.subs_rep.text() + "  # substitution_type\n")
+        fod_output.write(str(self.parent.ui.ndeg.value()) + "  # pla_type\n")
+        fod_output.write(self.parent.ui.plazcek_fit_range_min.text() + ", " + self.parent.ui.plazcek_fit_range_max.text() + "  # pla_range\n")
+        fod_output.write(self.parent.ui.ft_qrange.text() + "  # qrangeft\n")
+        fod_output.write(self.parent.ui.ff_rrange.text() + "  # fourier_range_r\n")
+        fod_output.write(self.parent.ui.ff_qrange.text() + "  # fourier_range_Q\n")
+        fod_output.write(self.parent.ui.secondary_scattering_ratio.text() + "  # secondary_scattering_ratio")
+        fod_output.close()
```

Comment:
Can we just call `fod_save_input` method here instead for the above?

Reply:
sure. will change it commit